### PR TITLE
Text2num release

### PR DESCRIPTION
**Pull request by rtxm**

### CODE REVIEW

File: _posts/2018-10-02-release-of-text2num-1.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title:  "Text2num version 1.0.0 released!"
+excerpt: "There already existed some python packages to convert numbers written in english into python numbers or their decimal
+digit representation, but there was nothing available for the french language. That's why we developped this library and shared it with the community."
```
Comment (n1k0) [inline _posts/2018-10-02-release-of-text2num-1.md]: Nit: *developed* (single p)

File: _posts/2018-10-02-release-of-text2num-1.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+---
+layout: post
+title:  "Text2num version 1.0.0 released!"
+excerpt: "There already existed some python packages to convert numbers written in english into python numbers or their decimal
```
Comment (n1k0) [inline _posts/2018-10-02-release-of-text2num-1.md]: Nit: *English* (same for *French*): language names are always capitalized.

File: _posts/2018-10-02-release-of-text2num-1.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+tags: python NLP
+---
+
+The output of speech-to-text systems are entirely made of words, without punctuation or capitalization. This makes visual scanning for numbers quite cumbersome,
+especially in transcriptions of real life dialogues as they also contain a lot of gibberish words — like « heu, ben, bah… » — and the syntax or
+grammar is not always right. Besides that, text mining tools and techniques often, if not always, expect numbers to be in decimal digit representation.
+
+That's why we decided to add a transformation pass to our speech-to-text engine in order to convert all spoken numbers into their digit spelling.
+
+Here at Allo-Media, we are fond of Open Source Software. So we first looked at the state of the art of python libraries for parsing words into numbers. There were some for the english language, but we didn't find any for french. Therefore, we decided to build our own library and contribute it back to the community.
```
Comment (n1k0) [inline _posts/2018-10-02-release-of-text2num-1.md]: Nit: *Python* (capitalized), and maybe add a link to python.org

File: _posts/2018-10-02-release-of-text2num-1.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+842 pommes, 25 chiens, 1003 chevaux, 12698 clous.
+95 vaut 95. On tolère l'absence de tirets avant les unités : 76 vaut 76.
+Nombres en série : 12 15 004 20 52 103 52 31.
+Ordinaux: 5ème 3ème 21ème 100ème 1230ème.
+Décimaux: 12,99, 120,05 ; mais 60 02.
+```
+
+As you see, we support decimal numbers as well as ordinal numbers.
+
+The algorithm is quite robust and is based on the observation that big numbers are structured by powers of thousand in the language, each group following the same pattern hundreds-tens-units. It handles the alternate forms « *septante / soixante-dix* » and « *nonante / quatre-vingt-dix* » and offers a relaxed mode for making the dash ([1990 reform](https://fr.wikipedia.org/wiki/Rectifications_orthographiques_du_fran%C3%A7ais_en_1990#Les_modifications_apport%C3%A9es)) optional.
```
Comment (n1k0) [inline _posts/2018-10-02-release-of-text2num-1.md]: Nit: *same pattern hundreds-tens-units*: unclear

Comment (n1k0) [reply _posts/2018-10-02-release-of-text2num-1.md]: Nit: maybe mention that nonante is the Belgian variant of quatre-vingt dix in France

File: _posts/2018-10-02-release-of-text2num-1.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+The project is hosted on GitHub : [https://github.com/allo-media/text2num](https://github.com/allo-media/text2num).
+
+Enjoy!
```
Comment (n1k0) [inline _posts/2018-10-02-release-of-text2num-1.md]: Maybe add a word about the licensing policy and the contribution process? Last, a link to the readthedocs documentation might be useful as well.

File: _posts/2018-10-02-release-of-text2num-1.md (hunk 3 above)
Comment (vjousse) [inline _posts/2018-10-02-release-of-text2num-1.md]: Shouldn't we give examples of what is already available for english? As a reader, this is an information I would like to know :)

File: _posts/2018-10-02-release-of-text2num-1.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+That's why we decided to add a transformation pass to our speech-to-text engine in order to convert all spoken numbers into their digit spelling.
+
+Here at Allo-Media, we are fond of Open Source Software. So we first looked at the state of the art of python libraries for parsing words into numbers. There were some for the english language, but we didn't find any for french. Therefore, we decided to build our own library and contribute it back to the community.
+
+Here are two samples of what you can expect from it:
+
+```python
+>>> from text_to_num import text2num
+>>> text2num('nonante-cinq')
```
Comment (vjousse) [inline _posts/2018-10-02-release-of-text2num-1.md]: Maybe adding the classical 'quatre-vingt quinze' before would avoid confusion like: "This is a belgian parser!" :)

File: _posts/2018-10-02-release-of-text2num-1.md (hunk 5 above)
Comment (merwan) [reply _posts/2018-10-02-release-of-text2num-1.md]: +1 for @n1k0 comment.
It would also be worth adding that the project is available on PyPI.

**Comment by rtxm** [comment]
I made quite a lot of changes in the second half of the post, so if you could check it again, I would be glad.

File: _posts/2018-10-02-release-of-text2num-1.md [hunk 7]
```diff
@@ -10,16 +10,32 @@ tags: python NLP
...
+- French has some pecularities like « *quatre-vingt-dix-neuf* » vs « *nonante-neuf* ».
+
+So we started a linguistic parser from scratch that is able to identify numbers and correctly isolate contiguous ones in a sequence. Moreover, we wanted it to be able
+to parse different flavors of french (e.g. *soixante-dix* and *septante* for 70, etc…).
+
+If you are interested in linguistics, *septante* for 70 and *nonante* for 90 are used in Belgium, Switzerland, Luxembourg, Aosta Valley, Jersey French and to a lesser extend in French regions of Savoie, Franche-Compté and even sometimes in Lorraine and Provence (source [Wikipedia](https://fr.wikipedia.org/wiki/70_(nombre)#Linguistique)). The rest of the French speaking world uses respectively *soixante-dix* and *quatre-vingt-dix*. The usage of *Huitante* and *octante* instead of *quatre-vingts* is [more restricted yet](https://fr.wikipedia.org/wiki/80_(nombre)#Huitante).
+
+As french spelling is a touchy topic, we also introduced a relaxed mode for making the dash from the [1990 spelling reform](https://fr.wikipedia.org/wiki/Rectifications_orthographiques_du_fran%C3%A7ais_en_1990#Les_modifications_apport%C3%A9es) optional.
```
Comment (n1k0) [inline _posts/2018-10-02-release-of-text2num-1.md]: Nit: *French* (cap F)

File: _posts/2018-10-02-release-of-text2num-1.md [hunk 8]
```diff
@@ -10,16 +10,32 @@ tags: python NLP
...
+We could have ported the [Word2number](https://pypi.org/project/word2number/) library, but it has some flaws:
+
+- It is unable to detect by itself the bounds of a number expression;
+- its algorithm is weak (ex: `w2n.word_to_num('hundred five fifty') == 550`);
+- French has some pecularities like « *quatre-vingt-dix-neuf* » vs « *nonante-neuf* ».
+
+So we started a linguistic parser from scratch that is able to identify numbers and correctly isolate contiguous ones in a sequence. Moreover, we wanted it to be able
+to parse different flavors of french (e.g. *soixante-dix* and *septante* for 70, etc…).
+
+If you are interested in linguistics, *septante* for 70 and *nonante* for 90 are used in Belgium, Switzerland, Luxembourg, Aosta Valley, Jersey French and to a lesser extend in French regions of Savoie, Franche-Compté and even sometimes in Lorraine and Provence (source [Wikipedia](https://fr.wikipedia.org/wiki/70_(nombre)#Linguistique)). The rest of the French speaking world uses respectively *soixante-dix* and *quatre-vingt-dix*. The usage of *Huitante* and *octante* instead of *quatre-vingts* is [more restricted yet](https://fr.wikipedia.org/wiki/80_(nombre)#Huitante).
```
Comment (n1k0) [inline _posts/2018-10-02-release-of-text2num-1.md]: Nit: *more restricted yet*: just `more restricted` sounds slightly better and idiomatic to me.

**Pull request merged**